File: Arrays/TwoSum.py
nums = [-1, -2, -3, -4,-5]
target = -8

nums_copy = []
indices = []

for number in nums:
    nums_copy.append(number)

for number in nums_copy:
    number_index = nums.index(number)
    other_num = target - number
    if (other_num in nums) and (nums.index(other_num) != number_index):
        indices.append(number_index)
        indices.append(nums.index(other_num))
    

print(indices)

File: Arrays/TwoSum.java
//-------------Problem: 1---------------

import java.util.*;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hashMap = new HashMap<>();

        for (int index = 0; index < nums.length; index++) {
            int value = nums[index];
            int complement = target - value;

            if (hashMap.containsKey(complement)) {
                return new int[]{hashMap.get(complement), index};
            }

            hashMap.put(value, index);
        }

        return new int[0]; 
    };
}
